recipes-kernel/kernel-tests/kernel-test-fbomb-files/fbomb.c

```diff
@@ -0,0 +1,407 @@
+// SPDX-License-Identifier: LGPL-2.1-only
+// Copyright (c) 2020 National Instruments.
+
+#define _GNU_SOURCE
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdarg.h>
+#include <string.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <err.h>
+#include <errno.h>
+#include <sched.h>
+#include <time.h>
+#include <sys/types.h>
+#include <sys/syscall.h>
+#include <sys/sysinfo.h>
+#include <linux/futex.h>
+#include <limits.h>
+#include <getopt.h>
+#include <ctype.h>
+
+#define TEST_CPU	1
+#define NSEC_PER_USEC	1000ULL
+#define NSEC_PER_SEC	1000000000ULL
+#define NOISE_THREADS	64
+
+#define FUTEX_TID_MASK	0x3fffffff
+
+typedef uint32_t futex_t;
+
+static futex_t cond;
+static futex_t lock1 = 0;
+static futex_t lock2 = 0;
+static pthread_barrier_t start_barrier;
+static long long int test_duration = 60;
+
+enum {
+        OPT_HELP = 1,
+        OPT_DURATION,
+};
+
+static struct option opt_long[] = {
+        {"help", no_argument, NULL, OPT_HELP},
+        {"duration", optional_argument, NULL, OPT_DURATION},
+        {NULL, 0, NULL, 0}
+};
+
+const char *opt_short = "d:h";
+
+static void usage(char* comm)
+{
+        printf("Usage:\n"
+               "%s <options>\n"
+	       "\t-d <time> --duration=<time>	Test duration in seconds\n"
+	       "\t\t\t\t\t(m, h, d modifiers can be used to specify minutes, hours, or days)\n"
+               "\t-h --help                       Display this help\n",
+               comm);
+}
+
+static int parse_options(int argc, char *argv[])
+{
+        int c;
+        int index = 0;
+	char *endptr = NULL;
+
+        for (;;) {
+                c = getopt_long(argc, argv, opt_short, opt_long, &index);
+                if (c < 0)
+                        break;
+                switch (c) {
+                        case 'd':
+                        case OPT_DURATION:
+                                test_duration = strtoll(optarg, &endptr, 10);
+				if (optarg == endptr)
+					goto err_exit;
+				switch (tolower(*endptr)) {
+					case 'm':
+						/* minutes */
+						test_duration *= 60;
+						break;
+					case 'h':
+						/* hours */
+						test_duration *= 60 * 60;
+						break;
+					case 'd':
+						test_duration *= 60 * 60 * 24;
+						/* days */
+						break;
+					default:
+						fprintf(stderr, "warning: unrecognized qualifier for test duration '%c'\n", *endptr);
+				}
+                                break;
+                        case '?':
+                        case 'h':
+                        case OPT_HELP:
+				goto err_exit;
+                }
+        }
+
+        return EXIT_SUCCESS;
+
+  err_exit:
+	usage(argv[0]);
+	return EXIT_FAILURE;
+}
```

Comment:
I see there is a `PASS: test_fbomb\n` line output directly from the test. Is this function an appropriate place to express a `FAIL: test_fbomb\n` line?

Reply:
I guess, but this is more of a failure of using the test correctly not an actual failure detected by the test (which we cannot detect within the test itself bc. the kernel will OOPS). Anyway it's not a big deal to add a print here so I'll do it.

Reply:
OK, I've cleaned-up the error exit paths and added a FAIL print. See fix-up commit.